compute/cloud-client/instances/src/create_firewall_rule.php

```diff
@@ -77,13 +76,14 @@ function create_firewall_rule(string $projectId, string $firewallRuleName, strin
     //Create the firewall rule using Firewalls Client.
     $operation = $firewallsClient->insert($firewallResource, $projectId);
 
-    // Wait for the create operation to complete using a custom helper function.
-    // @see src/wait_for_operation.php
-    $operation = wait_for_operation($operation, $projectId);
-    if (empty($operation->getError())) {
+    // Wait for the operation to complete.
+    $operation->pollUntilComplete();
+    if ($operation->operationSucceeded()) {
+        // $result = $operation->getResult(); // Optionally get operation result
         printf('Created rule %s.' . PHP_EOL, $firewallRuleName);
     } else {
-        printf('Firewall rule creation failed!' . PHP_EOL);
+        $error = $operation->getError();
+        printf('Firewall rule creation failed: %s' . PHP_EOL, $error);
```

Comment:
Likewise, `getError` is not a string, but an object (which cannot be cast into a string). So this needs to be more like:
```suggestion
        printf('Firewall rule creation failed: %s' . PHP_EOL, $error->getMessage());
```

Reply:
Fixed, although I didn't find an easy way to test it as it requires the LRO to fail. Trying simple bad calls like removing not existing VM or adding VM with the same name doesn't get here as it throws an ApiException with the initial API call (LRO is not started).

Reply:
I can't think of any way to mock or trigger an error like this

Reply:
So we'll probably need to leave it as is. Resolving this conversation.